src/main/java/org/mskcc/cbio/oncokb/web/rest/SlackController.java

```diff
@@ -92,12 +94,31 @@ public ResponseEntity<String> approveUser(@RequestParam("payload") String action
                                     userDTO = updatedUserDTO.get();
                                 }
                             }
+                        
+                            boolean apiAccessRequested = userDTO.getAdditionalInfo() != null && userDTO.getAdditionalInfo().getApiAccessRequest() != null && userDTO.getAdditionalInfo().getApiAccessRequest().isRequested();
+                            if (userDTO.getLicenseType() != LicenseType.ACADEMIC || apiAccessRequested) {
+                                Set<String> userDTOAuthorities = userDTO.getAuthorities();
+                                userDTOAuthorities.add(AuthoritiesConstants.API);
+                            }
+
                             Optional<UserDTO> updatedUser = userService.approveUser(userDTO, false);
                             if (updatedUser.isPresent() && updatedUser.get().isActivated()) {
                                 mailService.sendApprovalEmail(userDTO);
                             }
                         }
                         break;
+                    case APPROVE_USER_FOR_API_ACCESS:
+                        boolean apiAccessRequested = userDTO.getAdditionalInfo() != null && userDTO.getAdditionalInfo().getApiAccessRequest() != null && userDTO.getAdditionalInfo().getApiAccessRequest().isRequested();
+                        if (userDTO.getLicenseType() != LicenseType.ACADEMIC || apiAccessRequested) {
+                            Set<String> userDTOAuthorities = userDTO.getAuthorities();
+                            userDTOAuthorities.add(AuthoritiesConstants.API);
+                        }
+
+                        Optional<UserDTO> updatedUser = userService.updateUser(userDTO);
+                        if (updatedUser.isPresent() && updatedUser.get().isActivated()) {
+                            mailService.sendApiAccessApprovalEmail(userDTO);
+                        }
```

Comment:
Here though, we should make sure only send email if the user does not have API ROLE already, so we don't send the same email multiple times.

Reply:
This should only be triggered when the approve button is clicked in slack. The message is only sent to slack if the user has requested API access. The user can only request API access if they do not have ROLE API. I think we can do the check, but is it necessary?

Reply:
I do think it's necessary. The scenario of using the approve button may change, adding an additional check makes sense here.

Reply:
I think the current implementation doe snot prevent sending the email multiple times? We should record whether `userDTO` has ROLE_API before `updateUserWithRoleApiIfRequested`. Then use that together with the current if conditions. @bprize15